asserts/snap_asserts.go

```diff
@@ -110,6 +117,8 @@ func (snapdcl *SnapDeclaration) Prerequisites() []*Ref {
 	}
 }
 
+var validAppCommand = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$")
```

Comment:
I imagine mapping is not that nice though because you would still need to make the snap build pg-dump to map it back to pg_dump, that seems awkward